components/calendar/src/tests/extrema.rs

```diff
@@ -0,0 +1,50 @@
+use crate::Date;
+use crate::cal::*;
+use crate::Ref;
+
+#[test]
+fn check_extrema() {
+  // Minimum and maximum dates allowed in ECMA-262 Temporal.
+  let min_date_iso = Date::try_new_iso(-271821, 4, 19).unwrap();
+  let max_date_iso = Date::try_new_iso(275760, 9, 13).unwrap();
+
+  // Assertion failure:
+  // > Month should be in range of u8! Value -1428 failed for RD RataDie(-99280838)
+  {
+    let cal = Chinese::new();
+    let min_date = min_date_iso.to_calendar(Ref(&cal));
+    let max_date = max_date_iso.to_calendar(Ref(&cal));
+
+    println!("min.year = {:?}, max.year = {:?}", min_date.year(), max_date.year());
+  }
+
+  // Assertion failure:
+  // > Month should be in range of u8! Value -1428 failed for RD RataDie(-99280838)
+  {
+    let cal = Dangi::new();
+    let min_date = min_date_iso.to_calendar(Ref(&cal));
+    let max_date = max_date_iso.to_calendar(Ref(&cal));
+
+    println!("min.year = {:?}, max.year = {:?}", min_date.year(), max_date.year());
+  }
+
+  // Assertion failure:
+  // > assertion failed: Date::try_new_observational_islamic_date(y, m, d, IslamicObservational::new_always_calculating()).is_ok()
+  {
+    let cal = HijriSimulated::new_mecca();
+    let min_date = min_date_iso.to_calendar(Ref(&cal));
+    let max_date = max_date_iso.to_calendar(Ref(&cal));
+
+    println!("min.year = {:?}, max.year = {:?}", min_date.year(), max_date.year());
+  }
+
+  // Assertion failure:
+  // > assertion failed: Date::try_new_ummalqura_date(y, m, d, IslamicUmmAlQura::new_always_calculating()).is_ok()
```

Comment:
UAQ is not an astronomical calendar and it doesn't have this constructor.

Reply:
We should still test that it doesn't panic.

Reply:
Yeah but it feels like overkill to test all the calendars. These are the iffy ones, some iffier than others.

Reply:
In #4917 we have an example of UAQ panicking at some point in the past. That's justification enough for me to want a test here.

Reply:
that calendar has been rewritten since then. if the test is not panicking before you add the safe range, there's no point in adding it

Reply:
or when you set the safe range to `RataDie::MIN..RataDie::MAX`

Reply:
And we might change the implementation again. The calendar is of a type where this is _possible_, as is evidenced by this occurring in the past. It would be nice to find more testcases here but that doesn't mean we shouldn't at least have the boundary test.

Reply:
The purpose of this test is not to test that the code in this PR is doing something. The purpose of this test is to test that large Temporal-allowed dates work without panicking in the calendars that have calendrical debug assertions. We should ideally test more dates.

In the future if we moved the chinese calendar over to the pinqi approximation I would still wish to continue testing it here.

Reply:
by that logic we might change the implementation of every calendar, or add debug assertions to any calendar. then just add these tests to any_calendar.rs which has tests for all calendars.

Reply:
I mean yeah but it's only a couple calendars that are insufficiently defined outside the modern range. I think there's a pretty clear destination here.


But I might just make this an AnyCalendar test. I hadn't done that mostly because I was trying to reuse anba's test, and also the current AnyCalendar test is kind of annoyingly merged. Might try and clean it up a bit.

Reply:
I added more calendars.

any_calendar.rs has a massive omnibus test that I do not think we should expand. A good refactor to do here would be to design a "for all calendars" test utility.

Until then, I added some more calendars. I do not think we need to exhaustively list them all. I do think adding extra tests protects us from future issues.